defaults/network-ents_test.go

```diff
@@ -0,0 +1,157 @@
+package defaults
+
+import (
+	"syscall"
+	"testing"
+
+	"github.com/docker/libentitlement/secprofile"
+	"github.com/docker/libentitlement/types"
+	"github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNetworkNoneEntitlementEnforce(t *testing.T) {
+	entitlementID := "network.none"
+
+	ociProfile := secprofile.NewOCIProfile(testSpec(), "test-profile")
+	require.Contains(t, DefaultEntitlements, entitlementID)
+
+	ent := DefaultEntitlements[entitlementID]
+
+	newProfile, err := ent.Enforce(ociProfile)
+	require.NoError(t, err, "Failed enforce while testing entitlement %s", entitlementID)
+
+	newOCIProfile, err := ociProfileConversionCheck(newProfile, NetworkNoneEntFullID)
+	require.NoError(t, err, "Failed converting to OCI profile while testing entitlement %s", entitlementID)
+
+	require.NotNil(t, newOCIProfile.OCI)
+	require.NotNil(t, newOCIProfile.OCI.Linux)
+
+	require.NotNil(t, newOCIProfile.OCI.Process.Capabilities)
+
+	capsToRemove := []types.Capability{CapNetAdmin, CapNetBindService, CapNetRaw, CapNetBroadcast}
+	require.True(t, capsBlocked(*newOCIProfile.OCI.Process.Capabilities, capsToRemove))
+
+	pathsToMask := []string{"/proc/pid/net", "/proc/sys/net", "/sys/class/net"}
+	for _, pathToMask := range pathsToMask {
+		require.Contains(t, newOCIProfile.OCI.Linux.MaskedPaths, pathToMask)
+	}
+
+	nsToAdd := []specs.LinuxNamespaceType{specs.NetworkNamespace}
+	require.True(t, namespacesActivated(newOCIProfile.OCI.Linux.Namespaces, nsToAdd))
+
+	require.NotNil(t, newOCIProfile.OCI.Linux.Seccomp)
+
+	syscallsToBlock := []types.Syscall{SysSocket, SysSocketpair, SysSetsockopt, SysGetsockopt, SysGetsockname, SysGetpeername,
+		SysBind, SysListen, SysAccept, SysAccept4, SysConnect, SysShutdown, SysRecvfrom, SysRecvmsg, SysRecvmmsg, SysSendto,
+		SysSendmsg, SysSendmmsg, SysSethostname, SysSetdomainname,
+	}
+	require.True(t, seccompSyscallsBlocked(*newOCIProfile.OCI.Linux.Seccomp, syscallsToBlock))
+
+	syscallsWithArgsToAllow := map[types.Syscall][]specs.LinuxSeccompArg{
+		SysSocket: {
+			{
+				Index: 0,
+				Op:    specs.OpEqualTo,
+				Value: syscall.AF_UNIX,
+			},
+			{
+				Index: 0,
+				Op:    specs.OpEqualTo,
+				Value: syscall.AF_LOCAL,
+			},
+		},
+	}
+	require.False(t, seccompSyscallsWithArgsBlocked(*newOCIProfile.OCI.Linux.Seccomp, syscallsWithArgsToAllow))
```

Comment:
in a follow-up PR: it would be awesome to have an integration test that runs a container similar to how you've demoed this functionality.

Reply:
This will very probably be done on the Moby side but this is the next thing on my todo-list.